Client/Frontend/Browser/TabsBar/TabsBarViewController.swift

```diff
@@ -157,15 +157,34 @@ class TabsBarViewController: UIViewController {
         reloadDataAndRestoreSelectedTab()
     }
     
+    func updateSelectedTabTitle() {
+        guard let selectedTabIndex = selectedTabIndexPath,
+              let tab = tabList[selectedTabIndex.row] else { return }
+        if let cell = collectionView.cellForItem(at: selectedTabIndex) as? TabBarCell {
+            cell.titleLabel.text = tab.displayTitle
+        }
+        
+    }
+    
     func reloadDataAndRestoreSelectedTab() {
         collectionView.reloadData()
         
-        if let tabManager = tabManager, let selectedTab = tabManager.selectedTab {
```

Comment:
No changes here, only extracted selectedTabIndexPath to a helper function, please double check though